packages/webapp/components/layouts/AccountLayout/index.tsx

```diff
@@ -1,22 +1,44 @@
-import React, { ReactElement, ReactNode, useContext } from 'react';
+import React, { ReactElement, ReactNode, useCallback, useContext } from 'react';
 import { PublicProfile } from '@dailydotdev/shared/src/lib/user';
 import AuthContext from '@dailydotdev/shared/src/contexts/AuthContext';
 import { NextSeoProps } from 'next-seo/lib/types';
 import Head from 'next/head';
 import { NextSeo } from 'next-seo';
+import { ProfileSettingsMenu } from '@dailydotdev/shared/src/components/profile/ProfileSettingsMenu';
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { disabledRefetch } from '@dailydotdev/shared/src/lib/func';
+import {
+  generateQueryKey,
+  RequestKey,
+} from '@dailydotdev/shared/src/lib/query';
+import { useViewSize, ViewSize } from '@dailydotdev/shared/src/hooks';
 import { getLayout as getMainLayout } from '../MainLayout';
-import SidebarNav from './SidebarNav';
 import { getTemplatedTitle } from '../utils';
+import SidebarNav from './SidebarNav';
 
 export interface AccountLayoutProps {
   profile: PublicProfile;
   children?: ReactNode;
 }
 
+export const navigationKey = generateQueryKey(
```

Comment:
Not urgent, but seeing it used in 2 places now would maybe make for a good hook?
I'm missing a bit of context here, why do we need react query for this?
Wouldn't simply state work?

Reply:
Because the back button is found not on where we initialized the sidebar.

Reply:
Hmm seems like usePersistentContext almost?
Anyway up to you but dedicated hook might be needed just to maintain it better? (also optional)

Reply:
Yeah, could've used the persistent context, though we don't need the extra usage of the idb storage. Maybe we should introduce a hook without it called `useQueryState`.

Reply:
Hmm ah right, yes think in current flows we indeed just use dedicated hooks right to solve this.
We have quite a few that do something like this.

Reply:
Let me know what you think of this https://github.com/dailydotdev/apps/pull/2867/commits/588741f91b43500a62ddb819c573be0f7b0707a1